src/App.tsx

```diff
@@ -1,14 +1,55 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todoId, setTodoId] = useState(0);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [filter, setFilter] = useState('all');
+
+  useEffect(() => {
+    getTodos()
+      .then(todo => {
+        setTodos(todo);
+      });
+  }, []);
+
+  const preaperedSearchQuery = searchQuery.toLowerCase();
+
+  const visibleTodo = todos.filter(todo => {
+    const searchingTodo = todo.title.toLowerCase()
+      .includes(preaperedSearchQuery);
+
+    const swithCase = (value: string) => {
+      switch (value) {
+        case 'all':
+          return todo;
+
+        case 'active':
+          return todo.completed === true;
+
+        case 'completed':
+          return todo.completed === false;
+        default:
+          break;
+      }
+
+      return value;
+    };
+
+    const filteredTodo = swithCase(filter);
```

Comment:
it's something strange.

you create by every iteration swithCase function accepts some magic "value"

after this you call this function and pass selected filter option to it

after this you put result of this function to "filterTodo" variable, but this function returns not todo, but boolean value.

i dont understand nothing.

lets refactor this.

you dont need to create any functions here

just write switch case here that return boolean value for every todo, based on todo.completed and isSearchQueryMatched values